src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,76 @@
+package duke;
+
+import java.io.Serializable;
+import java.time.DayOfWeek;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a task with a deadline.
+ * Extends the base Task class and includes additional information about the deadline.
+ */
+
+public class Deadline extends Task implements Serializable {
+
+    /**
+     * The deadline date and time
+     */
+    protected LocalDateTime by;
+
+    /**
+     * The day of the week if the deadline is specified as a day (e.g., Monday).
+     */
+    protected DayOfWeek dayOfWeek;
+
+    /**
+     * Constructor for the Deadline class.
+     *
+     * @param description The description of the deadline task.
+     * @param by          The deadline in string format.
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        parseDeadline(by.trim());
+    }
+
+    /**
+     * Parses the deadline string to determine the deadline date and time or day of the week.
+     *
+     * @param by The deadline in string format.
+     */
+    private void parseDeadline(String by) {
+        try {
+
+            DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
+            this.by = LocalDateTime.parse(by, dateTimeFormatter);
+
+        } catch (DateTimeParseException e) {
+
+            try {
+                // If parsing as specific date/time format fails, try parsing as day of the week
+                DateTimeFormatter dayOfWeekFormatter = DateTimeFormatter.ofPattern("EEEE");
```

Comment:
Consider using constants for the date formats and declaring them as private static final fields at the top of the class. Doing so would make the method more generalised and if you needed to change the format, you would only need to change the constants instead of going to each method where it's used and changing it.

Example:
private static final String DATE_TIME_FORMAT = "d/M/yyyy HHmm";
private static final String DAY_OF_WEEK_FORMAT = "EEEE";